go_covert_lib/client/src/App.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from 'react';
+import Button from 'react-bootstrap/Button';
+import FormControl from 'react-bootstrap/FormControl';
+import InputGroup from 'react-bootstrap/InputGroup';
+import './styles.css';
+
+/**
+ * IMPORTANT NOTE: For styling, refer to https://getbootstrap.com/docs/4.0/utilities/position/
+ */
+
+const App = () => {
+  const [textToSend, setTextToSend] = useState('');
+  const [channels, setChannels] = useState([]);
+  let ws; // The websocket connection to the server
+
+  const sendConfig = () => {
+    const cmd = JSON.stringify({ OpCode: 'config' });
+    ws.send(cmd);
+  };
+
+  useEffect(() => {
+    // Matches just the "127.0.0.1:8080" portion of the address
+    const addressRegex = /[a-zA-Z0-9.]+:[\d]+/g;
+    ws = new WebSocket(`ws://${window.location.href.match(addressRegex)[0]}/api/ws`);
+    ws.binaryType = 'arraybuffer';
+    ws.onopen = (event) => {
+      sendConfig();
+    };
+
+    ws.onerror = (event) => {
+      // TODO:
+    };
+
+    ws.onmessage = (event) => {
+      const msg = JSON.parse(event.data);
+      switch (msg.OpCode) {
+        case 'config':
```

Comment:
Make sure to check out test.html to see how channels and processors are handled differently.
The covert  channel display is always present, whereas you can add multiple processors. The inputs for each different processor and channel must be dynamically generated and change when the user sets a different select.

Reply:
Yup, I've been making sure to follow that for everything I'm building. Working on the dynamic generation in the next issue of this milestone.